libraries/libraries.mk

```diff
@@ -17,43 +17,46 @@ CXXSOURCES += $(LIBRARIES_PATH)/bsg_manycore_responder.cpp
 CXXSOURCES += $(LIBRARIES_PATH)/bsg_manycore_tile.cpp
 CXXSOURCES += $(LIBRARIES_PATH)/bsg_manycore_uart_responder.cpp
 
+HEADERS += $(LIBRARIES_PATH)/bsg_manycore.h
+HEADERS += $(LIBRARIES_PATH)/bsg_manycore_bits.h
+HEADERS += $(LIBRARIES_PATH)/bsg_manycore_config.h
+HEADERS += $(LIBRARIES_PATH)/bsg_manycore_cuda.h
+HEADERS += $(LIBRARIES_PATH)/bsg_manycore_elf.h
+HEADERS += $(LIBRARIES_PATH)/bsg_manycore_eva.h
 HEADERS += $(LIBRARIES_PATH)/bsg_manycore_loader.h
+HEADERS += $(LIBRARIES_PATH)/bsg_manycore_memory_manager.h
+HEADERS += $(LIBRARIES_PATH)/bsg_manycore_origin_eva_map.h
+HEADERS += $(LIBRARIES_PATH)/bsg_manycore_printing.h
+HEADERS += $(LIBRARIES_PATH)/bsg_manycore_request_packet_id.h
+HEADERS += $(LIBRARIES_PATH)/bsg_manycore_responder.h
 HEADERS += $(LIBRARIES_PATH)/bsg_manycore_tile.h
+
 HEADERS += $(LIBRARIES_PATH)/bsg_manycore_vcache.h
-HEADERS += $(LIBRARIES_PATH)/bsg_manycore_eva.h
 HEADERS += $(LIBRARIES_PATH)/bsg_manycore_mmio.h
 HEADERS += $(LIBRARIES_PATH)/bsg_manycore_errno.h
-HEADERS += $(LIBRARIES_PATH)/bsg_manycore_bits.h
-HEADERS += $(LIBRARIES_PATH)/bsg_manycore_elf.h
-HEADERS += $(LIBRARIES_PATH)/bsg_manycore_memory_manager.h
-HEADERS += $(LIBRARIES_PATH)/bsg_manycore_cuda.h
 HEADERS += $(LIBRARIES_PATH)/bsg_manycore_features.h
-HEADERS += $(LIBRARIES_PATH)/bsg_manycore_printing.h
-HEADERS += $(LIBRARIES_PATH)/bsg_manycore_config.h
 HEADERS += $(LIBRARIES_PATH)/bsg_manycore_coordinate.h
-HEADERS += $(LIBRARIES_PATH)/bsg_manycore.h
 HEADERS += $(LIBRARIES_PATH)/bsg_manycore_npa.h
 HEADERS += $(LIBRARIES_PATH)/bsg_manycore_epa.h
-HEADERS += $(LIBRARIES_PATH)/bsg_manycore_origin_eva_map.h
 HEADERS += $(LIBRARIES_PATH)/bsg_manycore_packet.h
 HEADERS += $(LIBRARIES_PATH)/bsg_manycore_response_packet.h
 HEADERS += $(LIBRARIES_PATH)/bsg_manycore_request_packet.h
 HEADERS += $(LIBRARIES_PATH)/bsg_manycore_fifo.h
-HEADERS += $(LIBRARIES_PATH)/bsg_manycore_request_packet_id.h
-HEADERS += $(LIBRARIES_PATH)/bsg_manycore_responder.h
 
 CHEADERS   += $(HEADERS)
 CXXHEADERS += $(HEADERS)
 
 OBJECTS += $(patsubst %cpp,%o,$(CXXSOURCES))
 OBJECTS += $(patsubst %c,%o,$(CSOURCES))
-$(OBJECTS): INCLUDES  = -I$(SDK_DIR)/userspace/include
-$(OBJECTS): INCLUDES += -I$(LIBRARIES_PATH)
-$(OBJECTS): INCLUDES += -I$(HDK_DIR)/common/software/include 
-$(OBJECTS): INCLUDES += -I$(AWS_FPGA_REPO_DIR)/SDAccel/userspace/include 
+
+$(OBJECTS): INCLUDES  = -I$(LIBRARIES_PATH)
+$(OBJECTS): INCLUDES += -I$(SDK_DIR)/userspace/include
+$(OBJECTS): INCLUDES += -I$(HDK_DIR)/common/software/include
+$(OBJECTS): INCLUDES += -I$(AWS_FPGA_REPO_DIR)/SDAccel/userspace/include
 $(OBJECTS): CFLAGS    = -std=c11 -fPIC -D_GNU_SOURCE $(INCLUDES)
 $(OBJECTS): CXXFLAGS  = -std=c++11 -fPIC -D_GNU_SOURCE $(INCLUDES)
-$(OBJECTS): LDFLAGS   = -lfpga_mgmt
+$(OBJECTS): LDFLAGS   = -lfpga_mgmt -fPIC
+$(OBJECTS): % : $(HEADERS)
```

Comment:
not sure why you pattern match here...

Reply:
Oh, because I experimented with `$(OBJECTS): %.o : %.h`

Reply:
Fixed